src/overlays/actors/ovl_En_Zf/z_en_zf.c

```diff
@@ -82,115 +152,2281 @@ static ColliderQuadInit D_80B4A204 = {
+            EnZf_SetupSlash(this);
+        } else {
+            func_80B483E4(this, globalCtx);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zf/func_80B462E4.s")
+void EnZf_SetupJumpBack(EnZf* this) {
+    Animation_Change(&this->skelAnime, &gZfJumpingAnim, -1.0f, 3.0f, 0.0f, 2, -3.0f);
+    this->unk_3F0 = 0;
+    this->hopAnimIndex = 1;
+    this->action = ENZF_ACTION_JUMP_BACK;
+    this->actor.velocity.y = 15.0f;
+    this->actor.speedXZ = -15.0f;
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_RIZA_JUMP);
+    EnZf_SetupAction(this, EnZf_JumpBack);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zf/func_80B463E4.s")
+void EnZf_JumpBack(EnZf* this, GlobalContext* globalCtx) {
+    if ((this->unk_3F0 != 0) && (this->actor.world.pos.y <= this->actor.floorHeight)) {
+        this->actor.world.pos.y = this->actor.floorHeight;
+        this->hopAnimIndex = 0;
+        this->actor.velocity.y = 0.0f;
+        this->actor.speedXZ = 0.0f;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zf/func_80B46A24.s")
+    if (SkelAnime_Update(&this->skelAnime)) {
+        if (this->unk_3F0 == 0) {
+            Animation_Change(&this->skelAnime, &gZfLandingAnim, 3.0f, 0.0f, 17.0f, 2, -3.0f);
+            this->unk_3F0 = 10;
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_RIZA_JUMP);
+        } else if ((globalCtx->gameplayFrames % 2) != 0) {
+            func_80B483E4(this, globalCtx);
+        } else {
+            func_80B45384(this);
+        }
+    }
+
+    if ((globalCtx->state.frames & 0x5F) == 0) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_RIZA_CRY);
+    }
+}
+
+void EnZf_SetupStunned(EnZf* this) {
+    if ((this->actor.bgCheckFlags & 1) && ((this->actor.velocity.y == 0.0f) || (this->actor.velocity.y == -4.0f))) {
+        this->actor.speedXZ = 0.0f;
+        this->hopAnimIndex = 0;
+    } else {
+        this->hopAnimIndex = 1;
+    }
+
+    if (this->damageEffect == ENZF_DMGEFF_ICE) {
+        this->iceTimer = 36;
+    } else {
+        Animation_PlayOnceSetSpeed(&this->skelAnime, &gZfKnockedBackAnim, 0.0f);
+    }
+
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_GOMA_JR_FREEZE);
+    this->action = ENZF_ACTION_STUNNED;
+    EnZf_SetupAction(this, EnZf_Stunned);
+}
+
+void EnZf_Stunned(EnZf* this, GlobalContext* globalCtx) {
+    s16 angleToWall;
+
+    if (this->actor.bgCheckFlags & 2) {
+        this->actor.speedXZ = 0.0f;
+    }
+
+    if (this->actor.bgCheckFlags & 1) {
+        if (this->actor.speedXZ < 0.0f) {
+            this->actor.speedXZ += 0.05f;
+        }
+        this->hopAnimIndex = 0;
+    }
+
+    if ((this->actor.colorFilterTimer == 0) && (this->actor.bgCheckFlags & 1)) {
+        if (this->actor.colChkInfo.health == 0) {
+            EnZf_SetupDie(this);
+        } else if ((this->actor.params != ENZF_TYPE_DINOLFOS) || !EnZf_ChooseAction(globalCtx, this)) {
```

Comment:
Looking through again, I wonder if a lot of the `if (!EnZf_DodgeRanged(...))`s should actually be `if (EnZf_DodgeRanged(...)) { return; }`? It would certainly improve the indentation, and it happens in a lot of these actionFuncs